backend/wrapped/views.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from django.db import transaction
+from wrapped.models import Page, IndividualStat, GlobalStat
+from django.core.exceptions import ValidationError
+from django.http import HttpResponse
+from django.contrib.auth import get_user_model
+from rest_framework import generics, viewsets
+from rest_framework.decorators import action
+from django.shortcuts import get_object_or_404
+from .serializers import PageSerializer
+
+User = get_user_model()
+
+
+class GeneratePage(viewsets.ModelViewSet): 
```

Comment:
get the user through `request.user`, and make sure you have to be authenticated to use this route. see our other views.

Also I don't think ModelViewSet is the right viewset to use here.

Also call this and associated urls `Pages`. We're not really generating anything (generating implies side effects). We're just serializing data already in the db: we are returning `Page`s filtered for the given semester, serialized in a particular way.

Reply:
Actually now that we are adding a Semester model, this should be thought of as returning data associated with 1 particular semester. 

(And in the serializer for semester it turns it into a list of pages each of which can be serialized with the page serializer and so on...)